### PR TITLE
[REF] Refactor `performance_metrics` into `benchmarking.metrics` and use a local version of `sphinx-remove-toctrees`

### PR DESCRIPTION
**Pull request by MatthewMiddlehurst**
Moves performance metric and refactors the segmentation metrics into a single file.

Updates the benchmarking API file. While doing this I noticed `sphinx-remove-toctrees` had broke, See https://github.com/executablebooks/sphinx-remove-toctrees/issues/9. This PR introduces a fixed version locally.

### CODE REVIEW

File: docs/api_reference.md [hunk 1]
```diff
@@ -20,7 +20,6 @@ api_reference/data_format
 api_reference/datasets
 api_reference/distances
 api_reference/networks
-api_reference/performance_metrics
```
Comment (SebastianSchmidl) [inline docs/api_reference.md:23]: Where is the documentation now?

**Review by SebastianSchmidl — CHANGES_REQUESTED** [review]
I did not check whether there are other places still referencing the old metrics. Should be fine, though.

Are the metrics part of the public API? If yes, we should include them in the API reference.

File: docs/api_reference.md (hunk 1 above)
Comment (MatthewMiddlehurst) [reply docs/api_reference.md:23]: So far with these big module changes I have waited for it to settle down a bit before going into the docs, but this is the last change I wanted to make to benchmarking really.

Will update it under `benchmarking` and make sure the rest of the module is there as well

Comment (SebastianSchmidl) [reply docs/api_reference.md:23]: Makes sense, thank you!

File: docs/api_reference/benchmarking.rst [hunk 2]
```diff
@@ -3,14 +3,119 @@
 Benchmarking
 ============
 
-.. automodule:: aeon.benchmarking
-    :no-members:
-    :no-inherited-members:
+The ``aeon.benchmarking`` module contains tools and utilities for comparing and
+evaluating and time series models, loading stored results, and calculating performance
+metrics for a variety of tasks.
```
Comment (SebastianSchmidl) [inline docs/api_reference/benchmarking.rst:8]: This sentence has probably too many `and`s. I don't get it.

File: docs/api_reference/benchmarking.rst [hunk 3]
```diff
@@ -3,14 +3,119 @@
...
+
+.. currentmodule:: aeon.benchmarking.metrics.segmentation
+
+.. autosummary::
+    :toctree: auto_generated/
+    :template: function.rst
+
+    count_error
+    hausdorff_error
+    prediction_ratio
```
Comment (SebastianSchmidl) [inline docs/api_reference/benchmarking.rst:121]: What about the `aeon.benchmarking.metrics.stats` module containing
- check_friedman
- nemenyi_test
- wilcoxon_test

IMO, those are not metrics. I would move them to `aeon.benchmarking.stats` and also add them to the documentation here.

**Review by SebastianSchmidl — CHANGES_REQUESTED** [review]
Thanks for fixing the sphynx-plugin. I think the PR is still fine in size.

File: docs/api_reference/benchmarking.rst (hunk 3 above)
Comment (MatthewMiddlehurst) [reply docs/api_reference/benchmarking.rst:121]: good catch. made the suggested change

**Review by SebastianSchmidl — APPROVED** [review]

**Pull request merged**